### PR TITLE
remove solana-program from solana-feature-set

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
solana-feature-set can compile much faster by avoiding a solana-program dependency

#### Summary of Changes
Remove solana-program by replacing it with the crates it re-exports, and by replacing the Slot and Epoch type aliases with u64s

This branches off #2348 so that needs to be merged first

### CODE REVIEW

File: log-analyzer/Cargo.toml [hunk 1]
```diff
@@ -13,6 +13,7 @@ edition = { workspace = true }
 byte-unit = { workspace = true }
 clap = { version = "3.1.5", features = ["cargo"] }
 serde = { workspace = true }
+serde_derive = { workspace = true }
```
Comment (joncinque) [inline log-analyzer/Cargo.toml:16]: Unless I'm missing something, this change should be done in a separate PR

**Review by joncinque — COMMENTED** [review]
For this PR, if there are ever changes to `Slot` or `Epoch`, we wouldn't want to miss an update in this crate, so could we use `Slot` and `Epoch` from the solana-clock crate?

Otherwise, it all looks good to me!

File: log-analyzer/Cargo.toml (hunk 1 above)
Comment (kevinheavey) [reply log-analyzer/Cargo.toml:16]: it's a fix for something that only broke in this PR (idk why)

**Comment by kevinheavey** [comment]
> For this PR, if there are ever changes to `Slot` or `Epoch`, we wouldn't want to miss an update in this crate, so could we use `Slot` and `Epoch` from the solana-clock crate?

Done, we're already using solana-sdk-macro here so the extra compile time isn't that much

File: log-analyzer/Cargo.toml (hunk 1 above)
Comment (kevinheavey) [reply log-analyzer/Cargo.toml:16]: I think it's because solana-log-analyzer has an indirect dependency on solana-feature-set, and before this PR that was causing the "derive" feature of serde to be activated

**Review by joncinque — APPROVED** [review]
Looks great, thanks!
